modules/fargate/fargate.tf

```diff
@@ -0,0 +1,44 @@
+# EKS Fargate Pod Execution Role
+
+data "aws_iam_policy_document" "eks_fargate_pod_assume_role" {
+  count = var.create_eks ? 1 : 0
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["eks-fargate-pods.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "eks_fargate_pod" {
+  count              = var.create_eks ? 1 : 0
+  name               = format("%s-fargate", var.cluster_name)
```

Comment:
The cluster and worker roles are created using `name_prefix` by default. This allows for the same name cluster to be created in different regions as IAM is global. I have no idea if anybody actually does that. There is also the complexity around `name_prefix` vs `name` and `var.workers_role_name`. Do we want to replicate that here? @barryib

Reply:
This is the only open topic to resolve now. How would you like to handle the naming here @barryib ?